pymodbus/server/simulator/http_server.py

```diff
@@ -372,10 +372,10 @@ def build_html_calls(self, params: dict, html: str) -> str:  # pragma: no cover
         for function in DecodePDU(True).list_function_codes():
             selected = (
                 "selected"
-                if function.function_code == self.call_monitor.function
+                if function == self.call_monitor.function
                 else ""
             )
-            function_codes += f"<option value={function.function_code} {selected}>function code name</option>"
+            function_codes += f"<option value={function} {selected}>function code name</option>"
```

Comment:
This is wrong, you are now building a list of object pointers and not function codes !